pkg/api/rest/types.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rest
```

Comment:
Should be to be the default implementation of the mutation logic for each type.  Once we fold registries into rest we can move these into rest.  They don't have to be here, but logic was small enough so far.


> On Jan 29, 2015, at 8:58 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/api/rest/types.go:
> 
> > +Copyright 2014 Google Inc. All rights reserved.
> > +
> > +Licensed under the Apache License, Version 2.0 (the "License");
> > +you may not use this file except in compliance with the License.
> > +You may obtain a copy of the License at
> > +
> > +    http://www.apache.org/licenses/LICENSE-2.0
> > +
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package rest
> What's the purpose of this file? A comment would be useful.
> 
> —
> Reply to this email directly or view it on GitHub.
>